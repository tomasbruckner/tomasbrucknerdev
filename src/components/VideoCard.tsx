import React from 'react';
import { Card, CardContent, Typography } from '@material-ui/core';

interface Props {
  videoUrl: string;
  videoHeadline: string;
}

const VideoCard = ({ videoUrl, videoHeadline }: Props) => {
  return (
    <Card
      style={{
        minWidth: 275,
        marginTop: '48px',
        marginBottom: '48px',
      }}
    >
      <CardContent>
        <Typography variant="h5" component="h2" style={{ marginBottom: '16px' }}>
          {videoHeadline}
        </Typography>
        <iframe
          title={videoHeadline}
          width="100%"
          height="440"
          style={{ borderWidth: 0 }}
          src={videoUrl}
        />
      </CardContent>
    </Card>
  );
};

export default VideoCard;
