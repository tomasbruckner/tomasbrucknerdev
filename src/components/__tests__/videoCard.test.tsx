import React from 'react';
import VideoCard from '../VideoCard';
import renderer from 'react-test-renderer';

it('VideoCard renders correctly', () => {
  const tree = renderer
    .create(<VideoCard videoHeadline="testVideoHeadline" videoUrl="testVideoUrl" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
