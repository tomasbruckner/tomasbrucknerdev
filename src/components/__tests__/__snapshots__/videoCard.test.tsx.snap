// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root makeStyles-h2-2 MuiTypography-h5"
    >
      testVideoHeadline
    </h2>
    <iframe
      className="makeStyles-iframe-3"
      height="440"
      src="testVideoUrl"
      title="testVideoHeadline"
      width="100%"
    />
  </div>
</div>
`;
